### PR TITLE
Prepare release v0.7.0

### LINKED ISSUE #625: Add Image Cropping Support to imgcompress
## New Image Crop Feature for imgcompress

I wanted to build a feature that makes imgcompress more useful for everyday image editing.

imgcompress already supports image compression, conversion, and local AI background removal, but one important feature was missing: **image cropping**.

With the new crop feature, you can select the exact area of an image you want to keep before exporting, compressing, converting, or using background removal.

Thank you for all the support. Reaching over **41k Docker pulls** is amazing, and it motivates me to keep improving imgcompress in my free time.

## What the new crop feature does

The new image crop tool lets you:

- Crop images directly in imgcompress

- Select a custom crop area

- Use common aspect ratios

- Adjust the crop before saving

- Combine cropping with local AI background removal

- Continue with image compression or conversion after cropping

### 1. Import an image

Import an image as usual, then click the new **Crop** button.

Image

### 2. Select the crop area

The new crop widget opens. Select the part of the image you want to keep, then click **Save Crop**.

Image

### 3. Export the cropped image

After saving the crop, imgcompress applies the crop to the image. You can then continue with compression, conversion, or local AI background removal.

Image

## Why this helps

This feature makes imgcompress more useful as a complete local image optimization tool. You can now crop, compress, convert, and remove backgrounds without needing another app.

The new image crop feature is planned for the next imgcompress release.

## Image credit
Image by <a href="https://pixabay.com/users/iam_sam-4717682/?utm_source=link-attribution&utm_medium=referral&utm_campaign=image&utm_content=8255456">Samuel Kamau</a> from <a href="https://pixabay.com//?utm_source=link-attribution&utm_medium=referral&utm_campaign=image&utm_content=8255456">Pixabay</a>

### LINKED ISSUE #627: Can not run Dockerfile install pnpm as it blocked "unrs-resolver@1.11.1"
## What Happened?

I was currently testing a new Dockerfile for this repository, then this error jump out:

Image

---

It was `CI=true` environment ENV of `pnpm` blocking script exec of `sharp@0.34.5` and `unrs-resolver@1.11.1`.

In `packages.json`, I confirmed that the front-end allows `onlyBuiltDependencies` for `sharp`: 

Image

But where does the `unrs-resolver@1.11.1` come from? The Dockerfile I am working is basically has no different than the current one:

Image

## Why does it need attention?

One package suddenly want to exec script out of nowhere (but not in the whitelist) is not a normal behavior, especially It want to do that right now, not sooner. The fact that `npm` ecosystem was so screw up with `axios`, `bitwarden` really make me doubt about this abnormal.

So with my work on Dockerfile and container's entrypoint + healthcheck scripts, this package management conflict is blocking my work as I have no privilege to modify front-end's `packages.json`.

## Recommendation

Please check this `unrs-resolver@1.11.1` package, and notify me if It is safe to add to `onlyBuiltDependencies` whitelist, so I can continue my work.

Cheers,
Aleksandr.

### LINKED ISSUE #641: imgcompress-chan: a tiny helper for the boring stuff 💛
I sometimes have repetitive tasks on this repo, and I don't have endless time for the boring stuff. So I built [@imgcompress-chan](https://github.com/apps/imgcompress-chan) to help out. 💛

Here she is in action when I called her on a fork PR — she politely declines and tells the contributor how to repair it themselves:

Image

She's so cute, uwu.

And here she is doing the real job — a failed Dependabot dependency update with a desynced pnpm lockfile. One /chan-fix and she handles it:

Image

Now that imgcompress is growing it is the best time to get her as a new contributer, even though she is a custom bot please be kind to her. From now on I made sure just the author @karimz1 can call her but in the future I might give her more responsibility she just started haha.

### PR DESCRIPTION
**Pull request by karimz1**
> ⚠️ **Heads up: this RC will take longer than planned.** Scope has grown to include broader automated coverage and an internal architecture cleanup. Details below.

---

## Expanded test coverage

Building on the existing `ImgCompress` suite to reduce reliance on manual verification. New coverage areas:

- **Unverified format regression** — previously supported formats are validated automatically, no manual test files needed each cycle
- **Feature flag matrix** — all flag combinations are now exercised automatically
- **Edge-case scenarios** — known tricky paths that previously required manual verification

## Why passing tests aren't enough

This change touches enough internal logic that green tests alone aren't sufficient confidence. The codebase has well-known edge cases that aren't yet covered by the automated suite — those need a manual pass before tagging an RC.

## Motivation — internal refactor

This PR also includes a cleanup of the internal architecture in the `ImgCompress` solution. The goals:

- Enforce consistent use of **domain DTOs** — no more returning raw dictionaries where a typed object is the right call
- Eliminate hardcoded values throughout the solution
- Keep clean architecture principles consistent as the feature set and contributor base grows

This was always "good enough as an MVP" — dictionaries worked fine early on. But with more features, more contributors, and a growing user base, it made sense to give the internals a proper pass. All existing tests remain unchanged; only the internal Python logic has been abstracted further.

Here is an preview of the Test Matrix currently working on
image

## What's shipping in this release

- Image crop feature #625
- Changes from #626 — integrated and tested together before release

## Remaining work

- [x] Manual validation of edge cases not covered by automated tests
- [x] Integrate and test with #626 — big combined release, everything tested together before tagging and ensured it works very stable.
- [x] Tag STAGE RC (nightly) once confidence is high
- [x] QA pass to validate that all features work.

> Please be patient — the new RC is on its way, just a bit later than expected. Appreciate your understanding.

### CODE REVIEW

File: backend/image_converter/config/loader.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    WebConfig,
+)
+from backend.image_converter.domain.web_workers import WebWorkerCount
+
+
+class ConfigError(RuntimeError):
+    pass
+
+
+_ENV_TRUTHY = frozenset({"true", "1", "yes", "on"})
```
Comment (karimz1) [inline backend/image_converter/config/loader.py:35]: I honestly need to double check it do I still need it ? let me check

File: backend/image_converter/config/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Backend Config
```
Comment (karimz1) [inline backend/image_converter/config/README.md]: well that is obsolete people can read code this had not much important info

File: backend/image_converter/core/image_conversion_processor.py [hunk 3]
```diff
@@ -230,6 +230,10 @@ def output_results(self, summary: ConversionSummary) -> None:
             )
 
             response_dict = {k: v for k, v in asdict(response).items() if v is not None}
+            # CLI `--json-output` payload: this is the program's machine-readable
```
Comment (karimz1) [inline backend/image_converter/core/image_conversion_processor.py]: dead comments

File: backend/image_converter/presentation/web/services/backend_diagnostics_service.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            "",
+            f"generated_at: {datetime.now(timezone.utc).isoformat(timespec='seconds')}",
+            f"process_id: {os.getpid()}",
+            f"temp_dir: {self.temp_dir}",
+            f"log_file: {self.log_path_provider()}",
+            f"storage_management_enabled: {self.storage_management_service.is_storage_management_enabled()}",
+            "",
+            "## Notes",
+            "- This file contains backend startup output, app-level logs, and unhandled exceptions captured by imgcompress.",
+            "- It only includes entries written after this backend process started.",
```
Comment (karimz1) [inline backend/image_converter/presentation/web/services/backend_diagnostics_service.py]: less comments would follow clean code

File: backend/image_converter/presentation/web/static_routes.py [hunk 5]
```diff
@@ -1,8 +1,22 @@
 import os
 from flask import Blueprint, jsonify, send_from_directory, abort, current_app
 
+from backend.image_converter.config import settings
+
 static_blueprint = Blueprint("static_blueprint", __name__)
 
+
+@static_blueprint.route("/config/runtime.json")
+def serve_runtime_config():
```
Comment (karimz1) [inline backend/image_converter/presentation/web/static_routes.py:14]: I could simplify it

File: backend/image_converter/bootstraper.py [hunk 6]
```diff
@@ -1,59 +1,79 @@
...
+        bufsize=1,
+    )
+    assert web_server_process.stdout is not None
+    for line in web_server_process.stdout:
+        # Verbatim relay: granian's child stdout is already formatted (its own
+        # timestamp, level, worker id). Forward unchanged so the parent's
+        # TeeStream captures one line per granian line in the log file. Do
+        # NOT wrap with Logger — that would prepend a second timestamp and
+        # ANSI colors, polluting both the log file and `docker logs`.
+        print(line, end="")
```
Comment (karimz1) [inline backend/image_converter/bootstraper.py]: use logger no print

File: backend/image_converter/bootstraper.py [hunk 7]
```diff
@@ -1,59 +1,79 @@
...
+            "backend.image_converter.presentation.web.server:app",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        text=True,
+        bufsize=1,
+    )
+    assert web_server_process.stdout is not None
+    for line in web_server_process.stdout:
+        # Verbatim relay: granian's child stdout is already formatted (its own
```
Comment (karimz1) [inline backend/image_converter/bootstraper.py]: dead comments

File: frontend/src/components/crop/CropLoadingPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+            opacity: 1;
+            transform: translateY(0);
+            filter: blur(0);
+          }
+        }
+        .crop-word-pop {
+          animation: crop-word-pop 3.8s cubic-bezier(0.2, 0.8, 0.2, 1) infinite both;
+          will-change: transform, opacity, filter;
+        }
+      `}</style>
```
Comment (karimz1) [inline frontend/src/components/crop/CropLoadingPanel.tsx]: own scss would be better no hard coding css 💯

File: healthcheck.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import urllib.request
```
Comment (karimz1) [inline healthcheck.py:1]: idk about as I have the health check also on the web ui trigger so that would be double I guess let me check later

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+	/var/run/docker.sock:/var/run/docker.sock \
+	aquasec/trivy:0.70.0@sha256:be1190afcb28352bfddc4ddeb71470835d16462af68d310f9f4bca710961a41e \
+	image \
+	--severity HIGH,CRITICAL \
+	--format table \
+	--output scan-result.log \
+	$(REGISTRY)/$(IMAGE):$(TAG)
+
+local_build:
+	@bash runLocalDockerBuildTester.sh
```
Comment (karimz1) [inline Makefile]: I need to test that especially with the privy as well as cloud build maybe cloud build is not needed because I use GitHub actions and it has no limits

**Review by karimz1 — COMMENTED** [review]
somme comments
